### PR TITLE
[Tokenomics] feat: add `mint_allocation_proposer` param to tokenomics module

### PR DESCRIPTION
**Pull request by bryanchriswhite**
## Summary

Add the `mint_allocation_proposer` module parameter to the tokenomics module.

## Issue

- `TODO_BETA`

## Type of change

Select one or more from the following:

- [x] New feature, functionality or library
- [ ] Consensus breaking; add the `consensus-breaking` label if so. See #791 for details
- [ ] Bug fix
- [ ] Code health or cleanup
- [ ] Documentation
- [ ] Other (specify)

## Testing

- [ ] **Documentation**: `make docusaurus_start`; only needed if you make doc changes
- [x] **Unit Tests**: `make go_develop_and_test`
- [ ] **LocalNet E2E Tests**: `make test_e2e`
- [x] **DevNet E2E Tests**: Add the `devnet-test-e2e` label to the PR.

## Sanity Checklist

- [x] I have tested my changes using the available tooling
- [x] I have commented my code
- [x] I have performed a self-review of my own code; both comments & source code
- [ ] I create and reference any new tickets, if applicable
- [ ] I have left TODOs throughout the codebase, if applicable

### CODE REVIEW

File: x/tokenomics/types/params.go [hunk 1]
```diff
@@ -19,16 +22,21 @@ func ParamKeyTable() paramtypes.KeyTable {
...
 // NewParams creates a new Params instance
 func NewParams(
-	mintAllocationDao float64,
+	mintAllocationDao,
+	mintAllocationProposer float64,
 ) Params {
 	return Params{
-		MintAllocationDao: mintAllocationDao,
+		MintAllocationDao:      mintAllocationDao,
+		MintAllocationProposer: DefaultMintAllocationProposer,
```
Comment (bryanchriswhite) [inline x/tokenomics/types/params.go:30]:
```suggestion
		MintAllocationProposer: mintAllocationProposer,
```

**Review by red-0ne — COMMENTED** [review]
Same as the previous allocation params PRs. Let's settle on the `float64` param type name. Otherwise LGTM.

**Review by red-0ne — APPROVED** [review]
Smaller changes this time.

Approving.

File: tools/scripts/params/tokenomics_mint_allocation_proposer.json [hunk 2]
```diff
@@ -0,0 +1,12 @@
+{
+  "body": {
+    "messages": [
+      {
+        "@type": "/poktroll.tokenomics.MsgUpdateParam",
+        "authority": "pokt10d07y265gmmuvt4z0w9aw880jnsr700j8yv32t",
+        "name": "mint_allocation_proposer",
+        "as_double": 0.05
```
Comment (bryanchriswhite) [inline tools/scripts/params/tokenomics_mint_allocation_proposer.json:8]:
```suggestion
        "as_float": 0.05
```

**Pull request merged**